src/js/shariff.js

```diff
@@ -257,9 +257,12 @@ module.exports = Shariff;
 // export Shariff class to global (for non-Node users)
 global.Shariff = Shariff;
 
-// initialize .shariff elements
-$('.shariff').each(function() {
-    if (!this.hasOwnProperty('shariff')) {
-        this.shariff = new Shariff(this);
-    }
+$(global.document).ready(function () {
```

Comment:
This is actually not about xhtml, but about race conditions. You may not access DOM until it's ready. And you never know when it is ready => Hence, your change is a dedicated bugfix.